src/postgres/commands/configurePostgresFirewall.ts

```diff
@@ -19,15 +19,22 @@ export async function configurePostgresFirewall(context: IActionContext, treeIte
         treeItem = <PostgresServerTreeItem>await ext.tree.showTreeItemPicker(PostgresServerTreeItem.contextValue, context);
     }
 
-    const ip: string = await publicIp.v4();
+    const options: vscode.ProgressOptions = {
+        location: vscode.ProgressLocation.Notification,
+        title: localize('gettingPublicIp', 'Getting public IP...')
+    };
+
+    const ip: string = await vscode.window.withProgress(options, async () => {
+        return await publicIp.v4();
```

Comment:
Can you check for other cases where we call this? I feel like there's at least one more